File: 3174.cpp
#include <iostream>

using namespace std;

int main() {
	
	int n, h, p = 0, b = 0, a = 0, m = 0, d = 0, i;
	string e, g;
	
	cin >> n;
	
	while(i != n) {
		cin >> e >> g;
		if(g == "bonecos") {
			cin >> h;
			b += h;
		}
		if(g == "arquitetos") {
			cin >> h;
			a += h;
		}
		if(g == "musicos") {
			cin >> h;
			m += h;
		}
		if(g == "desenhistas") {
			cin >> h;
			d += h;
		}
		i++;
	}
	p = b/8 + (a/4) + (m/6) + (d/12);
	cout << p << endl;
	return 0;
}

File: URI-Python/3174.py
bonecosH = 0
arquitetosH = 0
musicosH = 0
desenhistasH = 0

n = int(input())

for x in range(n):
  l1,l2,l3 = map(str,input().split())
  if l2 == "bonecos":
     bonecosH = bonecosH + int(l3)
  elif l2 == "arquitetos":
     arquitetosH = arquitetosH + int(l3)
  elif l2 == "musicos":
     musicosH = musicosH + int(l3)
  elif l2 == "desenhistas":
     desenhistasH = desenhistasH + int(l3)

bonecos = int(bonecosH/8)
arquitetos = int(arquitetosH/4)
musicos = int(musicosH/6)
desenhistas = int(desenhistasH/12)

print(bonecos + arquitetos + musicos + desenhistas)